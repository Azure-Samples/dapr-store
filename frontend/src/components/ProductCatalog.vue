<template>
  <div>
    <h1>All Products</h1>
    <product-list view-type="main" />
  </div>
</template>

<script>
import ProductList from './ProductList'

export default {
  name: 'ProductCatalog',

  components: {
    'product-list': ProductList
  }
}
</script>

<style scoped>

</style>
