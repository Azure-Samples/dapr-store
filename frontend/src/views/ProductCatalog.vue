<template>
  <div>
    <h1>All Products</h1>
    <product-list view-type="all" />
  </div>
</template>

<script>
import ProductList from '../components/ProductList'

export default {
  name: 'ProductCatalog',

  components: {
    'product-list': ProductList
  }
}
</script>

<style scoped>

</style>
