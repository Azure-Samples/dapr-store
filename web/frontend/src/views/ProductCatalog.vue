<!--
// ----------------------------------------------------------------------------
// Copyright (c) Ben Coleman, 2020
// Licensed under the MIT License.
//
// Dapr Store frontend - Show all products in the catalog
// ----------------------------------------------------------------------------
-->

<template>
  <div>
    <h1>All Products</h1>
    <error-box :error="error" />
    <product-list v-if="!error" :products="products" />
  </div>
</template>

<script>
import api from '../services/api'
import ProductList from '../components/ProductList'
import ErrorBox from '../components/ErrorBox'

export default {
  name: 'ProductCatalog',

  components: {
    'product-list': ProductList,
    'error-box': ErrorBox
  },

  data() {
    return {
      products: null,
      error: null,
    }
  },

  async mounted() {
    try {
      let resp = await api.productCatalog()
      if (resp && typeof resp === 'object') {
        this.products = resp
      } else {
        throw new Error('Failed to fetch products')
      }
    } catch (err) {
      this.error = err
    }
  },
}
</script>
