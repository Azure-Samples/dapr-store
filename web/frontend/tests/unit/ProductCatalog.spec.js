import { createLocalVue, mount } from '@vue/test-utils'
import flushPromises from 'flush-promises'
import BootstrapVue from 'bootstrap-vue'
const localVue = createLocalVue()
localVue.use(BootstrapVue)

import ProductCatalog from '@/views/ProductCatalog.vue'

jest.mock('@/services/api')

describe('ProductCatalog.vue', () => {
  it('renders products in catalog', async () => {
    const wrapper = mount(ProductCatalog, { localVue,
      propsData: { }, sync: true
    })

    await flushPromises()
    expect(wrapper).toMatchSnapshot()
  })
})