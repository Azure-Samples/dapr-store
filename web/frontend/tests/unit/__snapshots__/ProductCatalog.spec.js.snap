// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCatalog.vue renders products in catalog 1`] = `
<div>
  <h1>All Products</h1>
  <!---->
  <div>
    <!---->
    <div class="card">
      <!---->
      <!---->
      <div class="card-body">
        <!---->
        <!---->
        <div class="row">
          <div class="col"><a href="/product/prd1" target="_self" class="">
              <h4 class="card-title">
                Top&nbsp;Hat&nbsp;(6″)
              </h4>
            </a>
            <p class="card-text">
              Made from 100% wool and quite nice
              <br><br>
            <h4>£39.95</h4>
            </p> <a role="button" aria-disabled="true" tabindex="-1" href="#" target="_self" class="btn d-none d-md-inline disabled btn-primary disabled">
              <fa icon="shopping-cart"></fa>
              &nbsp; Add to Cart
            </a>
          </div>
          <div class="flex-grow-0 d-none d-md-block col">
            <div class="product-img">
              <!----> <a href="/product/prd1" target="_self" class=""><img src="/img/catalog/1.jpg"></a></div>
          </div>
        </div>
      </div>
      <!---->
      <!---->
    </div>
    <div class="card">
      <!---->
      <!---->
      <div class="card-body">
        <!---->
        <!---->
        <div class="row">
          <div class="col"><a href="/product/prd2" target="_self" class="">
              <h4 class="card-title">
                Black &amp; Gold Tie Set
              </h4>
            </a>
            <p class="card-text">
              100% hand made silk includes tie, pocket square and cufflinks
              <br><br>
            <h4>£18</h4>
            </p> <a role="button" aria-disabled="true" tabindex="-1" href="#" target="_self" class="btn d-none d-md-inline disabled btn-primary disabled">
              <fa icon="shopping-cart"></fa>
              &nbsp; Add to Cart
            </a>
          </div>
          <div class="flex-grow-0 d-none d-md-block col">
            <div class="product-img"><span class="onsale">On Sale</span> <a href="/product/prd2" target="_self" class=""><img src="/img/catalog/2.jpg"></a></div>
          </div>
        </div>
      </div>
      <!---->
      <!---->
    </div>
    <div class="card">
      <!---->
      <!---->
      <div class="card-body">
        <!---->
        <!---->
        <div class="row">
          <div class="col"><a href="/product/prd3" target="_self" class="">
              <h4 class="card-title">
                Paisley Cravat Ascot Tie
              </h4>
            </a>
            <p class="card-text">
              Black &amp; Red Paisley Cravat
              <br><br>
            <h4>£11.2</h4>
            </p> <a role="button" aria-disabled="true" tabindex="-1" href="#" target="_self" class="btn d-none d-md-inline disabled btn-primary disabled">
              <fa icon="shopping-cart"></fa>
              &nbsp; Add to Cart
            </a>
          </div>
          <div class="flex-grow-0 d-none d-md-block col">
            <div class="product-img">
              <!----> <a href="/product/prd3" target="_self" class=""><img src="/img/catalog/3.jpg"></a></div>
          </div>
        </div>
      </div>
      <!---->
      <!---->
    </div>
  </div>
</div>
`;
